src/itype.h

```diff
@@ -607,11 +607,18 @@ struct itype {
 
 // Includes food drink and drugs
 struct it_comest : itype {
+    friend class Item_factory;
```

Comment:
However I'm looking at merging this, I'll follow up with a PR of my own doing what I'm suggesting, and we can discuss the general issue there, this is a long term thing, and the nutrition thing is short-term.